src/plugins/data/public/query/query_string/language_service/query_result.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import './_recent_query.scss';
+
+import { BehaviorSubject } from 'rxjs';
+import { EuiButtonEmpty, EuiPopover, EuiText, EuiPopoverTitle } from '@elastic/eui';
+
+import React, { useState, useEffect } from 'react';
+
+export interface QueryStatus {
+  status: string;
+  message?: {
+    error?: string;
+    message?: string;
+    statusCode?: number;
+  };
+  time?: number;
+}
+
+export function QueryResult(props: { queryStatus$: BehaviorSubject<QueryStatus> }) {
+  const [isPopoverOpen, setPopover] = useState(false);
+  const [queryStatus, setQueryStatus] = useState<QueryStatus>({ status: 'ready' });
+  const onButtonClick = () => {
+    setPopover(!isPopoverOpen);
+  };
+
+  useEffect(() => {
+    const subscription = props.queryStatus$.subscribe((status) => {
+      setQueryStatus(status);
+    });
+    return () => subscription.unsubscribe();
+  }, [props.queryStatus$]);
+
+  if (queryStatus.status === 'ready') {
```

Comment:
how come ready? ready to me seems like it hasn't made a request yet

should it be `SUCCESS`?

Reply:
oh i see it's from https://github.com/abbyhu2000/OpenSearch-Dashboards/blob/90d27bbd081e945557e8f66241a4e2d2cfb7c272/src/plugins/discover/public/application/view_components/utils/use_search.ts#L37

we should add these tyeps to data plugin and then export it and compare the enums